CMakeLists.txt

```
@@ -87,6 +87,11 @@ if(${USE_Z3_DLOPEN})
 elseif (${Z3_FOUND})
   add_definitions(-DHAVE_Z3)
   message("Z3 SMT solver found. This enables optional SMT checking with Z3.")
+  if (TESTS AND NOT EMSCRIPTEN)
+    if ("${Z3_VERSION}" VERSION_LESS "4.8.12.0")
+      message(WARNING "SMT tests need Z3 4.8.12 or higher, SMT tests are expected to fail!")
```

It's not `Z3 4.8.12 or higher`. It's exactly 4.8.12. Newer versions won't work either.

```suggestion
      message(WARNING "SMTChecker tests require Z3 4.8.12 for all tests to pass. When using an older Z3 version you will need to use the --no-smt flag.")
```

---

> No, it's not like that. We don't want to support building with multiple versions. It's just that older versions are acceptable for `solc` but not `soltest` and for such versions you should be able to build just `solc` successfully but building `soltest` should error out.


I guess we can do that
```
if (TESTS AND NOT EMSCRIPTEN)
  finde_package(Z3,  4.8.12)
else
  finde_package(Z3,  4.8.0)
endif()
```

also I need to print an informative error message

---

done